### PR TITLE
[Utilities] add modify_constants for MOI.modify in MatrixOfConstraints

### LINKED ISSUE #1748: Implement modification of function constant for MatrixOfConstraints
This is needed for https://github.com/jump-dev/SCS.jl/issues/246

### PR DESCRIPTION
**Pull request by odow**
Closes #1748

### CODE REVIEW

File: src/Utilities/matrix_of_constraints.jl [hunk 1]
```diff
@@ -219,6 +219,23 @@ This function should be implemented to be usable as storage of constants for
 """
 function set_from_constants end
 
+"""
+    modify_constants(constants, row::Int, new_constant::T) where {T}
+    modify_constants(
+        constants,
+        rows::AbstractVector{Int},
```
Comment (odow) [inline src/Utilities/matrix_of_constraints.jl:226]:
```suggestion
        rows::AbstractVector{<:Integer},
```

File: src/Utilities/matrix_of_constraints.jl [hunk 2]
```diff
@@ -219,6 +219,23 @@ This function should be implemented to be usable as storage of constants for
 """
 function set_from_constants end
 
+"""
+    modify_constants(constants, row::Int, new_constant::T) where {T}
```
Comment (odow) [inline src/Utilities/matrix_of_constraints.jl:223]:
```suggestion
    modify_constants(constants, row::Integer, new_constant::T) where {T}
```

File: src/Utilities/matrix_of_constraints.jl [hunk 3]
```diff
@@ -652,3 +669,36 @@ function MOI.get(
...
+    row::Int,
+    new_constant::T,
+) where {T}
+    b[row] = new_constant
+    return
+end
+
+function modify_constants(
+    b::AbstractVector{T},
+    rows::AbstractVector{Int},
```
Comment (odow) [inline src/Utilities/matrix_of_constraints.jl:697]:
```suggestion
    rows::AbstractVector{<:Integer},
```

File: src/Utilities/matrix_of_constraints.jl [hunk 4]
```diff
@@ -652,3 +669,36 @@ function MOI.get(
...
+        modify_constants(model.constants, rows(model, ci), change.new_constant)
+    catch
+        throw(MOI.ModifyConstraintNotAllowed(ci, change))
+    end
+    return
+end
+
+function modify_constants(
+    b::AbstractVector{T},
+    row::Int,
```
Comment (odow) [inline src/Utilities/matrix_of_constraints.jl:688]:
```suggestion
    row::Integer,
```

File: src/Utilities/matrix_of_constraints.jl [hunk 5]
```diff
@@ -219,6 +219,23 @@ This function should be implemented to be usable as storage of constants for
...
+"""
+    modify_constants(constants, row::Integer, new_constant::T) where {T}
+    modify_constants(
+        constants,
+        rows::AbstractVector{<:Integer},
+        new_constants::AbstractVector{T},
+    ) where {T}
+
+Modify `constants` in-place to store `new_constant` in the `row` row, or
+This function returns an instance of the set `S` for which the constants where
```
Comment (blegat) [inline src/Utilities/matrix_of_constraints.jl:231]: It seems to me that the function returns `nothing`

**Comment by blegat** [comment]
Nice, this is also a blocker to rewrite https://github.com/osqp/OSQP.jl/blob/master/src/MOI_wrapper.jl with MatrixOfConstraints

File: src/Utilities/matrix_of_constraints.jl (hunk 5 above)
Comment (odow) [reply src/Utilities/matrix_of_constraints.jl:231]: Ha. It looks like I just didn't finish typing this

Comment (odow) [reply src/Utilities/matrix_of_constraints.jl]: Fixed

**Review by blegat — APPROVED** [review]

**Pull request merged**